### PR TITLE
DBZ-40 Snapshotting and connector restart

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -399,4 +399,8 @@ public boolean snapshotSchema() {
             return snapshotSchema;
         }
     }
+
+    protected Clock getClock() {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java]: Is this truly needed? I'd prefer if sub-classes stored the reference from within their own constructor, if they need it.

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java]: @gunnarmorling I tend to disagree here - the parent class is the part of the subclass so it is a duplication. It is probably not needed to add the getter and just change the vsibility of the field.

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java]: Ok, works for me, too.

File: debezium-core/src/main/java/io/debezium/schema/SchemaChangeEvent.java [hunk 2]
```diff
@@ -37,7 +37,8 @@ public SchemaChangeEvent(Map<String, ?> partition, Map<String, ?> offset, String
         this.offset = Objects.requireNonNull(offset, "offset must not be null");
         this.database = Objects.requireNonNull(database, "database must not be null");
         this.schema = Objects.requireNonNull(schema, "schema must not be null");
-        this.ddl = Objects.requireNonNull(ddl, "ddl must not be null");
+        // DDL is not mandatory
+        this.ddl = ddl;
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/schema/SchemaChangeEvent.java]: Ok, so you plan to solely rely on `TableChanges` for SQL Server. That's cool, I guess.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks, @jpechane!

**Pull request merged**